google/cloud/spanner/retry_policy.h

```diff
@@ -32,8 +33,23 @@ struct SafeGrpcRetry {
     return status.ok();
   }
   static inline bool IsTransientFailure(google::cloud::Status const& status) {
-    return status.code() == StatusCode::kUnavailable ||
-           status.code() == StatusCode::kResourceExhausted;
+    if (status.code() == StatusCode::kUnavailable ||
+        status.code() == StatusCode::kResourceExhausted) {
+      return true;
+    }
+    // Treat the unexpected termination of the gRPC connection as retryable.
+    // There is no explicit indication of this, but it will result in an
+    // INTERNAL status with one of the `kRetryableMessages`.
+    if (status.code() == StatusCode::kInternal) {
+      constexpr char const* kRetryableMessages[] = {
```

Comment:
Nit: Given that this is `IsTransientFailure()`, I might choose to call these `kTransientMessages`?